schema-nft-marketplace.graphql

```diff
@@ -0,0 +1,255 @@
+    name: String
+
+    " Collection symbol, mirrored from the smart contract. Leave null if not available. "
+    symbol: String
+
+    " Total supply of the collection, mirrored from the smart contract. "
+    totalSupply: BigInt!
+
+    " NFT Standard the collection uses. "
+    nftStandard: NftStandard!
+
+    " Royalty fee rate in percentage. E.g. 2.5% should be 2.5 "
+    royaltyFee: BigDecimal!
+
+    " Number of tokens listed for trade in the marketplace. "
+    listedCount: BigInt!
+
+    " Floor price (in ETH) "
+    floorPriceETH: BigDecimal!
+
+    " Cumulative trade volume (in ETH) "
+    cumulativeTradeVolumeETH: BigDecimal!
+
+    " Revenue that goes to the marketplace protocol, aka protocol fee. "
+    marketplaceRevenueETH: BigDecimal!
+
+    " Revenue that goes to creator, aka royalty fee. "
+    creatorRevenueETH: BigDecimal!
+
+    " Sum of marketplaceRevenue and creatorRevenue. "
+    totalRevenueETH: BigDecimal!
+
+    " Trade count of the collection on the marketplace. "
+    tradeCount: Int!
+
+    " Trades of the collection. "
+    trades: [Trade!]! @derivedFrom(field: "collection")
+}
+
+" Trades exist such as a combination of taker/order and bid/ask. "
+type Trade @entity {
+    " { maker txn hash }-{ taker txn hash } "
+    id: ID!
+
+    " Transaction hash of the maker. "
+    makerTransactionHash: String!
+
+    " Transaction hash of the taker. "
+    takerTransactionHash: String!
+
+    " Event log index. "
+    logIndex: Int!
+
+    " Block timestamp where the trade is executed. "
+    timestamp: BigInt!
+
+    " Block number where the trade is executed. "
+    blockNumber: BigInt!
+
+    " Collection involved "
+    collection: Collection!
+
+    " Token ID of the traded NFT. "
+    tokenId: BigInt!
+
+    " Price (in ETH) "
+    priceETH: BigDecimal!
+
+    " Stretegy that the trade is executed. "
+    strategy: SaleStrategy!
+
+    " Buyer account address "
+    buyer: String!
+
+    " Seller account address "
+    seller: String!
+}
+
+type MarketplaceDailySnapshot @entity {
+    " { Contract address }-{# of days since Unix epoch time} "
+    id: ID!
+
+    " Block number where the snapshot is taken. "
+    blockNumber: BigInt!
+
+    " Block timestamp when the snapshot is taken. "
+    timestamp: BigInt!
+
+    " Number of collections listed on the marketplace. "
+    collectionCount: Int!
+
+    " Cumulative trade volume (in ETH) "
+    cumulativeTradeVolumeETH: BigDecimal!
+
+    " Revenue that goes to the marketplace protocol, aka protocol fee. "
+    marketplaceRevenueETH: BigDecimal!
+
+    " Revenue that goes to creator, aka royalty fee. "
+    creatorRevenueETH: BigDecimal!
+
+    " Sum of marketplaceRevenueETH and creatorRevenueETH. "
+    totalRevenueETH: BigDecimal!
+
+    " Trade count of the all collections on the marketplace. "
+    tradeCount: Int!
+
+    " Cumulative unique traders "
+    cumulativeUniqueTraders: Int!
```

Comment:
does this make more sense to be `cumulativeUniqueOwners` or is that different from what you mean by `cumulativeUniqueTraders` (to me, this feels like this is anyone who has interacted with a collection on the marketplace which might be what we want to capture)?